doc/rewriting-services.md

```diff
@@ -0,0 +1,204 @@
+# Tips for rewriting legacy services
+
+## First, write some tests
+
+If service tests don’t exist for the legacy service, stop and write them first.
+It’s recommended to PR these separately. If there’s some test coverage, it’s
+probably fine to move right ahead and add more in the process. Make sure the
+tests are passing, though.
+
+## Organization
+
+1. When there’s a single legacy service that handles lots of different things
+(e.g. version, license, and downloads), it should be split into three separate
+service classes and placed in three separate files.
+`myservice-version.service.js`, `myservice-license.service.js`,
+`myservice-downloads.service.js`.
+
+2. When a service has different variants of basically the same thing, it’s okay
+to put them in the same service. For example, daily/weekly/monthly/total
+downloads can go in one badge, and star rating vs point rating vs rating count
+can go in one badge, and same with various kinds of detail about a pull request.
+The hard limit (as of now anyway) is *one category per service*.
```

Comment:
Is there a clearer word we can use than 'service' here? (maybe 'badge or 'class'?) We refer to `myservice-version.service.js` etc in the previous paragraph. Can we make it clearer that `myservice-version` (and so on) must belong to one category but `myservice` can contain badges belonging to multiple categories. In general we tend to use the terms 'badge' and 'service' a bit ambiguously, but it is important to be clear in materials aimed at new contributors.